### ISSUE TITLE
LambdaTestTool - serverless plugin

### ISSUE THREAD

**Issue by samueleresca** [aws/aws-lambda-dotnet#358]
Hi,
I'm planning to create a serverless framework plugin to run lambda locally using the LambdaTestTool project. Any opinion about that?

**Comment by normj** [comment]
To be clear you mean the open source framework [Serverless Framework](https://serverless.com/)?

I welcome contributions and curious what you are thinking. Can you provide anymore details on what you envision the plugin doing?

**Comment by samueleresca** [comment]
Hi @normj, yeah I mean the  [Serverless Framework](https://serverless.com/). I've seen that you are adding the loading implementation for the `serverless.yml` file in the following PR: #357  Do you have an ETA for that? 

Yesterday, I experienced a runtime error when I tried to run the test tool using the `lambda-test-tool-yaml` branch:

```
System.IO.FileLoadException : Could not load file or assembly 'YamlDotNet, Version=5.0.0.0, Culture=neutral, PublicKeyToken=null'. Error 2147500035 (Exception from HRESULT: 0x80004003 (E_POINTER))
```
Not sure is the error is related to my local setup.

About the serverless plugin,  My idea is to automate the process of running the LambdaTestTool and provide a way to run tests by configuring an inbound event an checking the response without using the web interface provided by LambdaTestTool, a sort of code-driven integration tests.

Samuele

**Comment by normj** [comment]
By any chance does your Lambda function that you are debugging use YamlDotNet? The testing tool is using the signed version which shouldn't interfere wiht your code if you are using the unsigned version.

I'm still on wondering if I should have put the yaml parsing in `Amazon.Lambda.TestTool.ExternalCommands` and done the YAML parsing out of process to avoid the potential assembly clash. 

It was on my thoughts for down the road to make to make it so you could pass all of the relevant info that is setup in the web UI to be done on the commandline to make the tool more automatable. Is that what you are thinking of adding?

**Comment by samueleresca** [comment]
The problem I've described:
`System.IO.FileLoadException : Could not load file or assembly 'YamlDotNet, Version=5.0.0.0, Culture=neutral, PublicKeyToken=null'. Error 2147500035 (Exception from HRESULT: 0x80004003 (E_POINTER))`
System.IO.FileLoadException : Could not load file or assembly 'YamlDotNet, Version=5.0.0.0, Culture=neutral, PublicKeyToken=null'. Error 2147500035 (Exception from HRESULT: 0x80004003 (E_POINTER))` was probably related to a wrong restore of the packages on my local dev machine.  About keeping the YAML parsing on a separate project can be very helpful.   
Yeah, it would be great to have a CLI to execute the tool. In that way, it is possible to integrate it in the CI testing step.

Finally, I've added another PR #364  to support another `serverless.yml` file format, let me know what you think.
